pageserve/test_routes.py

```diff
@@ -0,0 +1,141 @@
+"""Unit tests for pageserve service HTTP routes."""
+
+# Authors: mukobi
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from unittest import mock
+import flask
+import ast
+from app import app
+
+VALID_USER_IN_SESSION = {
+    "user_id": "42",
+    "name": "Ford Prefect"}
+
+
+class TestAuthenticateAndGetUser(unittest.TestCase):
+    """Test authentication endpoint POST /v1/authenticate."""
+
+    def assert_sessions_are_equal(self, first, second):
+        """Asserts the two sessions are the same.
+
+        Only compares the "user_id" and "name" fields, but ignores the
+        "gauth_token" field which is not returned from the users service
+        but is stored in the session by app.authenticate_with_users_service.
+        """
+        self.assertEqual(first["user_id"], second["user_id"])
+        self.assertEqual(first["name"], second["name"])
+
+    def setUp(self):
+        """Set up test client."""
+        app.config["TESTING"] = True  # propagate exceptions to test client
+        app.secret_key = "Dummy key used for testing the flask session"
+
+    def test_valid_authentication(self):
+        """GAuth token successfully authenticates."""
+        with app.test_request_context():
+            with app.test_client() as client:
+                with mock.patch("app.requests") as mock_requests:
+                    mock_response = mock.MagicMock()
+                    mock_response.status_code = 201
+                    mock_response.json.return_value = VALID_USER_IN_SESSION
+                    mock_response.content = str(VALID_USER_IN_SESSION)
+                    mock_requests.post.return_value = mock_response
+                    result = client.post("/v1/authenticate", data={
+                        "gauth_token": "I don't matter because requests is mocked"})
+                    self.assertEqual(result.status_code, 201)
+                    result_dict = ast.literal_eval(result.data.decode())
+                    self.assert_sessions_are_equal(
+                        result_dict, VALID_USER_IN_SESSION)
+                    # user stored in session
+                    self.assertEqual(len(flask.session), 1)
+                    self.assertEqual(
+                        flask.session["user"], VALID_USER_IN_SESSION)
+
+    def test_failed_authentication(self):
+        """GAuth token fails authentication."""
+        response_content = "Error: bad gauth_token."
```

Comment:
This style seems different from the style in the test above (i.e. define some constants before starting the context managers vs literals or global constants). Make them consistent (either style is fine).

Reply:
Done https://github.com/knative-portability/large-events/pull/37/commits/a74b1c700a7671fe772b22e0c8ed31a1d262a0ad